src/skip.js

```diff
@@ -75,7 +75,8 @@ class Skip extends BasePlugin {
   _setOutroData(outro: SkipPoint): void {
     if (typeof outro?.startTime === 'number') {
       if (typeof outro?.endTime !== 'number' || outro?.endTime === -1) {
-        outro.endTime = this.player.duration;
+        outro.endTime = this.player.duration - 1;
+        this.eventManager.listen(this.player, this.player.Event.DURATION_CHANGE, () => (this._outro.endTime = this.player.duration - 1));
```

Comment:
better to do it only once. so move _setOutroData from FIRST_PLAY to DURATION_CHANGE

Reply:
_setOutroData will executed every time duration change instead of being executed once, or i don’t got you

Reply:
i mean doinf this assingment `this._outro.endTime = this.player.duration - 1` only once

Reply:
It takes a while for the event to happen and there is no guarantee at all that it will happen that why i think we better assign it regardless of the event and not relay on it

Reply:
"It takes a while for the event to happen and there is no guarantee that it will happen at all" this is just because you listen to it after on `FIRST_PLAY`.
extract it from it you will see it's triggered and reliable

Reply:
the _initListeners called on load media and therefore it called after the first (and The only one event guaranteed)